### PR TITLE
Implement agent management 

### PR DESCRIPTION
**Pull request by imbyungjun**
- Implement agent list
- Implement agent detail

### CODE REVIEW

**Comment by donggyu04** [comment]
Can one of the admins verify this patch?

File: ngrinder-controller/src/main/java/org/ngrinder/agent/controller/AgentManagerApiController.java [hunk 1]
```diff
@@ -41,29 +48,233 @@
...
+			return true;
+		} else {
+			return agentRegion.startsWith(region + "_owned") || region.equals(agentRegion);
+		}
+	}
+
+	/**
+	 * Filter agent list using user authority and user id
+	 */
+	private boolean filterAgentByUserAuthorityAndId(Collection<? extends GrantedAuthority> authorities, String userId, String region, String agentRegion) {
```
[the remark this answers is not in the document]
Comment (imbyungjun) [reply ngrinder-controller/src/main/java/org/ngrinder/agent/controller/AgentManagerApiController.java]: That will be much clear. I'm gonna change it.

File: ngrinder-controller/src/main/java/org/ngrinder/agent/controller/AgentManagerApiController.java [hunk 2]
```diff
@@ -41,29 +48,233 @@
...
+	/**
+	 * Get the current performance of the given agent.
+	 *
+	 * @param ip agent ip
+	 * @param name agent name
+	 * @param region agent region
+	 *
+	 * @return json message
+	 */
+
```
[the remark this answers is not in the document]
Comment (imbyungjun) [reply ngrinder-controller/src/main/java/org/ngrinder/agent/controller/AgentManagerApiController.java]: I will fix it.

File: ngrinder-frontend/src/js/components/agent/List.vue [hunk 3]
```diff
@@ -175,12 +230,24 @@
                 this.selectedAgents = [];
             }
         }
+
+        get ids() {
+            return { params: { ids: this.selectedAgents.join(',') } };
+        }
```
Comment (imbyungjun) [inline ngrinder-frontend/src/js/components/agent/List.vue]: It does not need to be a computed property. Just `getParams() { ... }` looks better.

**Pull request merged**